### PR TITLE
[Preprint Withdrawals][Reviews][EMB-548][EMB-551] Preprint Withdrawal for Reviews.

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

This PR adds the functionality for preprint service moderators to see a list of withdrawal requests and to withdraw a preprint or to accept a withdrawal request (if there is one).

## Summary of Changes

- Notice entry for withdrawing a preprint is added to the action feed on the reviews index page.
  - The `action-feed-entry` component is modified.
- A new `Withdrawal Request` tab is added to allow moderators to see the list of withdrawal requests (the `withdrawals` view).
  - The `moderation-base` component is modified to add the `Withdrawal Request` tab to allow redirection to `withdrawals` view.
  - A new component `request-list-row` component is added to show each single withdrawal requests.
  - The `moderation-list` component is modified to use the newly added `request-list-row` component for showing withdrawal requests.
- The `preprint-detail` view is modified to allow moderators to accept/deny a withdrawal request, or to withdraw a published preprint directly when no withdrawal request exists. It is also changed to show a tombstone page for withdrawn preprints.
  - In `preprint-detail.js`,  `submitDecision()` is removed. Instead, two new actions `submitReviewsDecision()` and `submitRequestsDecision()` is added to handle the decision of two different types.
    - `submitReviewsDecision()` is the same as `submitDecison()`.
    - `submitRequestsDecision()` will make a new `preprint-request-action` instance.
  - The `preprint-status-banner` component is changed to take both the `submitReviewsDecision()` and `submitRequestsDecision()` actions so that moderators can submit differernt decisions depending on whether there is a withdrawal request attached to the current preprint.
- Miscellaneous CSS file and translation string file changes.

## Testing Notes

For pre-moderation:
pre-moderation

For post-moderation:
post-moderation

Be sure to test all possible state transitions as shown in the sketch above. Note that most state transtions are unidirectional execept for that between `Accepted` and `Rejected` for pre-moderated preprints.

## Ticket

https://openscience.atlassian.net/browse/IN-327
https://openscience.atlassian.net/browse/IN-329

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

**Comment by adlius** [comment]
Tests are failing as this is not using the version of `ember-osf` with the newly added models. Should be fixed once the eosf PR is merged and this PR updated to use the new version of eosf.

File: app/components/preprint-status-banner/template.hbs [hunk 1]
```diff
@@ -16,86 +20,158 @@
...
+                                <label local-class="decision-option">
+                                    {{radio-button value="rejected" groupValue=decision changed="decisionToggled"}}
+                                    <span class="p-l-sm">
+                                        {{t labelDecline}}
+                                        <p>{{t declineExplanation}}</p>
+                                    </span>
+                                </label>
+                                <label local-class="withdrawal-justification">
+                                    <span class="p-l-sm">
+                                        {{t 'components.preprintStatusBanner.withdrawalJustification'}}
```
Comment (alexschiller) [inline app/components/preprint-status-banner/template.hbs]: Looks a little funny if you you select reject and these words are there with no justification
screen shot 2018-09-13 at 12 25 31 pm

File: app/components/moderation-list-row/component.js [hunk 2]
```diff
@@ -74,11 +79,26 @@ export default Component.extend({
...
+    // translations for withdrawn on label
+    withdrawnOnLabel: computed('submission.dateWithdrawn', function() {
+        const i18n = this.get('i18n');
+        const [withdrawnDate, gtDayWithdrawn] = this.formattedDateLabel(this.get('submission.dateWithdrawn'));
+        const dayValue = gtDayWithdrawn ? 'gtDay' : 'ltDay';
+        const labels = withdrawnOnLabel[dayValue];
+        return i18n.t(labels, { timeDate: withdrawnDate, moderatorName: this.get('latestActionCreator') });
+    }),
+
+    icon: computed('submission', function() {
```
Comment (alexschiller) [inline app/components/moderation-list-row/component.js]: This is fine as a 1-liner
```javascript
    icon: computed('submission', function() {
        return this.get(`iconClass.${this.get('submission.reviewsState')}`);
    }),
```

File: app/components/moderation-base/component.js [hunk 3]
```diff
@@ -17,15 +17,20 @@ export default Component.extend({
...
+    tabs: computed('theme.{reviewableStatusCounts.pending,requestStatusCounts.pending}', function() {
         return [
             {
                 nameKey: 'global.submissions',
                 route: 'preprints.provider.moderation',
                 hasCount: true,
                 count: this.get('theme.reviewableStatusCounts.pending'),
             },
+            {
+                nameKey: 'global.withdrawalRequests',
```
Comment (alexschiller) [inline app/components/moderation-base/component.js:29]: This section might need some tweaking, things break into two lines at medium and very small sizes. 

I'm not sure I have a good solution recommendation. 
screen shot 2018-09-13 at 2 41 20 pm

File: app/components/moderation-base/component.js [hunk 4]
```diff
@@ -40,4 +45,6 @@ export default Component.extend({
             },
         ];
     }),
+
```
Comment (alexschiller) [inline app/components/moderation-base/component.js:48]: extra lines

**Review by alexschiller — COMMENTED** [review]
dropping comments so you have things to do. Was not completely done reviewing yet

File: app/components/moderation-list/template.hbs [hunk 5]
```diff
@@ -48,6 +62,22 @@
...
+                                {{request-list-row request=request}}
+                            {{/each}}
+                            <div class="pull-right text-right">
+                                {{pagination-pager
+                                    count=totalPages
+                                    current=page
+                                    change=(action pageChanged)
+                                }}
+                            </div>
+                        {{/if}}
```
Comment (jamescdavis) [inline app/components/moderation-list/template.hbs:80]: Not sure the logic is correct here. It seems like submissions and requests will both always display. Please review.

**Review by jamescdavis — CHANGES_REQUESTED** [review]
One thing and please update with develop.

File: app/components/moderation-list/template.hbs (hunk 5 above)
Comment (adlius) [reply app/components/moderation-list/template.hbs:80]: After looking at the code, here's what's happening. The `moderation-list` template is used in two routes: moderation (showing a list of preprint submissions) and withdrawal requests (showing a list of withdrawal requests). `isSubmissions` and `isRequests`  are used to distinguish between usage in these two different views. Therefore, `requests` will only be defined if `isRequests` is true, and `submissions` will be defined if `isSubmissions` is true.

**Review by jamescdavis — APPROVED** [review]

**Pull request merged**